esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6268,119 +6212,194 @@ void CWsDfuEx::exportRecordDefinitionBinaryType(const char *recordDefinition, Me
+        StringBuffer requestId = req.getRequestId();
+        bool returnTextResponse = req.getReturnTextResponse();
+
+        if (isEmptyString(fileName))
+             throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "DFUCreateFile: No Name defined.");
+        if (isEmptyString(clusterName))
+             throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "DFUCreateFile: No Cluster defined.");
+        if (isEmptyString(recordDefinition))
+             throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "DFUCreateFile: No ECLRecordDefinition defined.");
 
         StringBuffer userId;
-        Owned<IUserDescriptor> userDesc;
         context.getUserID(userId);
+        Owned<IUserDescriptor> userDesc;
         if (!userId.isEmpty())
         {
             userDesc.setown(createUserDescriptor());
             userDesc->set(userId.str(), context.queryPassword(), context.querySignature());
         }
 
+        ClusterType clusterType = getClusterTypeByClusterName(clusterName);
+        if (clusterType == NoCluster)
+             throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "DFUCreateFile: Cluster %s not found.", clusterName);
+
+        StringBuffer groupName;
+        getClusterGroupName(groupName, clusterName);
+        if (isEmptyString(groupName))
+             throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "DFUCreateFile: Group for cluster %s not found.", clusterName);
+        Owned<IGroup> group = queryNamedGroupStore().lookup(groupName.str());
+        if (!group)
+            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "DFUCreateFile: Failed to get Group %s.", groupName.str());
+
         CDfsLogicalFileName lfn;
         lfn.set(fileName);
-        StringBuffer tempFileName = lfn.get();
+        StringBuffer normalizedFileName = lfn.get();
+
+        checkLogicalName(normalizedFileName, userDesc, false, true, false, nullptr);
+
+        StringBuffer tempFileName = normalizedFileName;
+        tempFileName.append(".").append(dfuCreateUniqId++); // avoid potential clash if >1 creating file. One will succeed at publish time.
         tempFileName.append(DFUFileCreate_FileNamePostfix);
 
-        StringBuffer jobId = requestBase.getJobId();
-        if (jobId.isEmpty())
-            jobId.appendf("Create %s on %s", tempFileName.str(), clusterName);
+        if (requestId.isEmpty())
+            requestId.appendf("Create %s on %s", normalizedFileName.str(), clusterName);
 
-        Owned<IFileDescriptor> fileDesc = createFileDescriptor(tempFileName, clusterTypeEx, groupName, group);
-        fileDesc->queryProperties().setProp("@job", jobId);
+        Owned<IFileDescriptor> fileDesc = createFileDescriptor(tempFileName, clusterTypeString(clusterType, false), groupName, group);
+        fileDesc->queryProperties().setProp("@job", requestId);
         if (!userId.isEmpty())
             fileDesc->queryProperties().setProp("@owner", userId);
+        fileDesc->queryProperties().setProp("ECL", recordDefinition);
```

Comment:
perhaps out of scope, but is it worthwhile to normalize the recordDef before setting it as the ECL prop? From the JAPI perspective, the ECL prop is important to determine the file structure, but we end up having to support multiple styles of record definitions in ECL. Just a thought

Reply:
well if it is malformed the code that follows will fail and generate a ' Failed to parse ECL ..' error.
I'll move the setProp("ECL ) line below, to make it more clear.

Reply:
what I was after was normalization of the ECL used to declare the rec def.
I don't have examples on hand, but from memory, it appears we support multiple ways of declaring a rec def.
As mentioned above, this is perhaps out of scope, but worth having the conversation.

Reply:
ok, I don't think there's currently a way to convert the binary info to ECL (which I want for other reasons).
If we had that, then parsing to the rtlType (which will also validate it) and then converting to ECL - will be a way to normalize it.
I'll open a separate JIRA for this.